### PR TITLE
Digi implementation

### PR DESCRIPTION
**Pull request by peterus**

### CODE REVIEW

File: src/TaskRouter.cpp [hunk 1]
```diff
@@ -21,6 +24,7 @@ bool RouterTask::setup(System &system) {
   _beaconMsg = std::shared_ptr<APRSMessage>(new APRSMessage());
   _beaconMsg->setSource(system.getUserConfig()->callsign);
   _beaconMsg->setDestination("APLG01");
+  _beaconMsg->setPath("WIDE1-1");
```
Comment (peterus) [inline src/TaskRouter.cpp:27]: I think we need to create two messages, one for APRS-IS and one for the digi

File: src/TaskRouter.cpp [hunk 2]
```diff
@@ -31,28 +35,69 @@ bool RouterTask::setup(System &system) {
...
-    std::shared_ptr<APRSMessage> msg  = _fromModem.getElement();
-    String                       path = msg->getPath();
+    std::shared_ptr<APRSMessage> modemMsg  = _fromModem.getElement();
+    String path;
 
-    if (!(path.indexOf("RFONLY") != -1 || path.indexOf("NOGATE") != -1 || path.indexOf("TCPIP") != -1)) {
-      if (!path.isEmpty()) {
-        path += ",";
+    if (system.getUserConfig()->aprs_is.active && modemMsg->getSource() != system.getUserConfig()->callsign) {
+      std::shared_ptr<APRSMessage> aprsIsMsg = std::make_shared<APRSMessage>(*modemMsg);
```
Comment (peterus) [inline src/TaskRouter.cpp:42]: this is not a valid/good code. use the `modemMsg` directly or create a new copy with the `new` keyword.

File: src/TaskRouter.cpp [hunk 3]
```diff
@@ -31,28 +35,69 @@ bool RouterTask::setup(System &system) {
 bool RouterTask::loop(System &system) {
   // do routing
   if (!_fromModem.empty()) {
-    std::shared_ptr<APRSMessage> msg  = _fromModem.getElement();
-    String                       path = msg->getPath();
+    std::shared_ptr<APRSMessage> modemMsg  = _fromModem.getElement();
+    String path;
```
Comment (peterus) [inline src/TaskRouter.cpp:39]: do not define this object here if you just need it further down.

File: src/TaskRouter.cpp [hunk 4]
```diff
@@ -31,28 +35,69 @@ bool RouterTask::setup(System &system) {
...
+        logPrintlnD("APRS-IS: no forward => RFonly");
       }
-      msg->setPath(path + "qAR," + system.getUserConfig()->callsign);
+    } else {
+      if (!system.getUserConfig()->aprs_is.active)
+        logPrintlnD("APRS-IS: disabled");
 
-      _toAprsIs.addElement(msg);
+      if (modemMsg->getSource() == system.getUserConfig()->callsign)
+        logPrintlnD("APRS-IS: no forward => own paket erceived");
```
Comment (peterus) [inline src/TaskRouter.cpp:63]: typo

File: src/TaskRouter.cpp [hunk 5]
```diff
@@ -31,28 +35,69 @@ bool RouterTask::setup(System &system) {
...
+    }
+
+    if (system.getUserConfig()->digi.active && modemMsg->getSource() != system.getUserConfig()->callsign) {
+      std::shared_ptr<APRSMessage> digiMsg = std::make_shared<APRSMessage>(*modemMsg);
```
Comment (peterus) [inline src/TaskRouter.cpp:67]: same as in line 42

File: src/TaskRouter.cpp (hunk 2 above)
Comment (peterus) [reply src/TaskRouter.cpp:42]: make_shared is perfect, sorry for the confusion!

File: src/TaskRouter.cpp (hunk 5 above)
Comment (dg0tm) [reply src/TaskRouter.cpp:64]: ignoriert :)

File: src/TaskRouter.cpp (hunk 3 above)
Comment (dg0tm) [reply src/TaskRouter.cpp]: soved?

File: src/TaskRouter.cpp [hunk 6]
```diff
@@ -86,7 +86,7 @@ bool RouterTask::loop(System &system) {
 
     if (system.getUserConfig()->digi.beacon) {
       std::shared_ptr<APRSMessage> digiBeaconMsg = std::make_shared<APRSMessage>(*_beaconMsg);
-      digiBeaconMsg->setPath("WIDE1-1");
+      //digiBeaconMsg->setPath("WIDE1-1");
```
Comment (peterus) [inline src/TaskRouter.cpp:89]: please remove line, no commented code :)

Comment (dg0tm) [reply src/TaskRouter.cpp]: resolved by ee2f14b

**Pull request merged**